elements/includes/creative_commons.inc

```diff
@@ -72,269 +124,1089 @@ function xml_form_elements_creative_commons($element, &$form_state) {
+    return FALSE;
+  }
+  return 'license/' . $subpath . '/get';
+}
+
+/**
+ * Get a license from the CC API.
+ *
+ * @param string $path
+ *   The path from which to query.
+ * @param array $options
+ *   The http query array to attach to the query.
+ *
+ * @return SimpleXMLElement|FALSE
+ *   The response from the REST API as a SimpleXMLElement, or FALSE if request
+ *   failed.
+ *
+ * Note that there may be a bug in the CC API. Querying, e.g., 'fi'
+ * jurisdiction, 'sa' derivatives and 'y' or 'n' commercial, results in a
+ * CC-BY-SA/CC-BY-NC-SA 4.0 International license, instead of the corresponding
+ * 1.0 Finland version (which exists and is not deprecated).
+ * This behaviour, unfortunately, breaks our logic, so users will have to
+ * insert those licenses manually, and admins cannot set them as default.
+ */
+function xml_form_elements_creative_commons_query_api_get($path, $options = array()) {
+  return _xml_form_elements_creative_commons_query_api($path, $options);
+}
+
+/**
+ * Get license details from the CC API.
+ *
+ * @param string $license_uri
+ *   The license URI for which to query details.
+ *
+ * @return SimpleXMLElement|FALSE
+ *   The response from the REST API as a SimpleXMLElement, or FALSE if request
+ *   failed.
+ */
+function xml_form_elements_creative_commons_query_api_details($license_uri) {
+  return _xml_form_elements_creative_commons_query_api('details', array('license-uri' => $license_uri));
+}
+
+/**
+ * Helper function to query the CC API.
+ *
+ * @param string $path
+ *   The API path to query.
+ * @param array $query
+ *   The query.
+ *
+ * @return SimpleXMLElement|FALSE
+ *   The response from the REST API as a SimpleXMLElement, or FALSE if request
+ *   failed.
+ */
+function _xml_form_elements_creative_commons_query_api($path, $query = array()) {
+  $response = drupal_http_request(url(XML_FORM_ELEMENTS_CREATIVE_COMMONS_API_URL . $path, array('query' => $query)));
+  if ($response->code != 200) {
+    watchdog('xml_form_elements', 'The Creative Commons API responded with code %code.', array('%code' => $response->code), WATCHDOG_WARNING);
+    return FALSE;
+  }
+  return simplexml_load_string($response->data, 'SimpleXMLElement');
+}
+
+/**
+ * Ask the creative commons REST API for a 'by' license.
+ *
+ * Left as a function for use by community modules.
+ *
+ * @deprecated As of release 7.x-1.11, this function remains only for backwards
+ *   compatibilty. New code should use
+ *   xml_form_elements_creative_commons_query_api_get() instead.
+ *   Update your code, as this function might be removed.
+ *
+ * @param string $commercial
+ *   'y' to allow, 'n' to disallow.
+ * @param string $derivatives
+ *   'y' to allow, 'n' to disallow, 'sa' to share alike.
+ * @param string $jurisdiction
+ *   Legal jurisdiction code.
+ * @param bool $disabled
+ *   If TRUE, return FALSE; otherwise perform query.
+ *
+ * @return mixed
+ *   SimpleXMLElement the return from the REST API.
+ *   FALSE if failed request.
```

Comment:
"if failed request "... or $disabled == disabled? Crazy logic

Reply:
@DiegoPino note that this is a legacy function that we are deprecating now. Do you want a modification in the documenting comment (and, btw, it would be "or `$disabled == TRUE`")?

Reply:
legacy or not, it is marked as new lines added by this pull. Means you are contributing to this.

Reply:
@DiegoPino No worries, I can update that...